Appc 1035 develop po a standalone sdk

brunovarelaAptoide opened the pull request:
**What does this PR do?**

  - This PR has the POA standalone Developments and the fix of the handshake starting before verifying the campaign.
- Wallet communication to check the campaign.
- Logs inserts for debug and to check in with stage of the POA we are.

**Database changed?**

No

**Where should the reviewer start?**

- POAManager.java
- WalletUtils.java
- Repository Package

**How should this be manually tested?**

- Using Wallet Dev
- 4 scenarios : 

1º Wallet installed and Campaign Available
2º Wallet not installed and Campaign Available
3º Wallet not installed and Campaign not available
4º Wallet installed and Campaign Available

**What are the relevant tickets?**

Appc 1035 , APPC-1112

**Questions:**

New AIDL , and new Bind adress.

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

brunovarelaAptoide committed:
Changed the HandShake invocation

- Changed the HandShake invocation for POA to be done after checking for the existence of a campaign.

brunovarelaAptoide committed:
Wallet Validation when call campaignService

- Verification of the wallet installation to choose if the calls dirrectly the WS or goes through the Wallet

brunovarelaAptoide committed:
Added support for wallet service

-Removed comments
-WalletService Deactivated.

brunovarelaAptoide committed:
Created WalletConnection

- Created Wallet Connection Classes and Listenners.

brunovarelaAptoide committed:
Connection to Wallet Working

- Campaign mapped
- Changed properties

brunovarelaAptoide committed:
POA Standalone

- POA Standalone Working with all the requirements.

nunomonteiro commented on appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/network/AppCoinsClient.java:
```diff
@@ -22,15 +22,21 @@ public AppCoinsClient(String packageName, int versionCode, String serviceUrl,
 
   public void getCampaign(QueryParams queryParams, ClientResponseHandler clientResponseHandler) {
-    HTTPResponseHandler httpResponseHandler = new HTTPResponseHandler(clientResponseHandler);
+    CampaignResponseHandler CampaignResponseHandler =
```
var name should be lowercase

nunomonteiro commented on appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/network/listeners/CheckConnectivityResponseListener.java:
```diff
@@ -1,5 +1,5 @@
 
 public interface CheckConnectivityResponseListener {
-  void responseConnectivity(boolean value);
+  void responseConnectivity(boolean hasConnectivity);
```
well done!

nunomonteiro commented on appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/campaign/CampaignMapper.java:
```diff
@@ -53,5 +55,18 @@ private static String GetStringValue(String paramName, String response) {
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+    return new Campaign(new BigInteger("-1"), "");
```
Shouldn't we log an error here?
We might not have an exception and the campaign will be created with index -1 and fail silently

nunomonteiro commented on appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java:
```diff
@@ -351,4 +344,65 @@ private void initiateProofSending() {
+      stopProcess();
+    } else {
+      if (isWalletInstalled) {
+        walletIsInstalled(campaign);
```
Method name is not explicit of what's about to happen, it's just mirroring that bool that enables it to run. A better name would be startCampaign

nunomonteiro commented on appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java:
```diff
@@ -351,4 +344,65 @@ private void initiateProofSending() {
+      if (isWalletInstalled) {
+        walletIsInstalled(campaign);
+      } else {
+        walletIsNotInstalled();
```
Method name is not explicit of what's about to happen, it's just mirroring that bool that enables it to run. A better name would be promptWalletInstall

nunomonteiro commented on appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/WalletUtils.java:
```diff
@@ -7,48 +7,95 @@
 
-  private static void showWalletInstallDialog( final Activity activity,Context context,String message,final DialogVisibleListener dialogVisibleListener) {
+    if (act == null) {
+      return;
```
Should log error failure before returning

nunomonteiro commented on gradle.properties:
```diff
@@ -14,6 +14,14 @@ ROPSTEN_NETWORK_BACKEND_BASE_HOST=https://apichain-dev.blockchainds.com
 MAIN_NETWORK_BACKEND_BASE_HOST_WS=https://api.blockchainds.com
 ROPSTEN_NETWORK_BACKEND_BASE_HOST_WS=https://api-dev.blockchainds.com
 
+APPCOINS_ADVERTISEMENT_DEV_BIND_ACTION=com.appcoins.wallet.advertising.action.BIND
```
Shouldn't this have .dev. as well? I recall that we've talked about this but can't remember the conclusion

nunomonteiro reviewed (changes requested):
Please check inline comments

brunovarelaAptoide committed:
Pull request Changes

- Changes of pull request

brunovarelaAptoide committed:
Changes Notification

- Notification Working with install button.

brunovarelaAptoide committed:
Notifcation

- Install button going to Aptoide and if aptoide is not installed it goes to google play.
- Only shows the pop up the first time

brunovarelaAptoide committed:
Changes

-making notification work for android 25 < and 25 >

brunovarelaAptoide committed:
Code Cleaning

- Changes to cleanup and organize some code.

brunovarelaAptoide committed:
Removed Bug found in PR

- Removed not openning AppStore when click in notification in android versions bellow 26

brunovarelaAptoide committed:
Changes WalletUtils

- Changed the double intent send by SDK.

brunovarelaAptoide committed:
changes to meet Dev branch

- Change billing Wallet utils to meet the dev branch

brunovarelaAptoide committed:
Aptoide redirect remove

- Removed aptoide store redirection

JoaoRaimundo commented on appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/campaign/CampaignMapper.java:
```diff
@@ -53,5 +56,20 @@ private static String GetStringValue(String paramName, String response) {
+    }
+
+    Log.d(CampaignMapper.class.getName(), "No campaign is available.");
+    return new Campaign(new BigInteger("-1"), "");
```
The definition of an invalid campaign id should be obtained from the Campaign class itself and not defined hardcoded. Please use INVALID_CAMPAIGN field from Campaign class.

The defined log may come to be redundant since we use the same log in the manager that processes the campaign id:
```
if (!campaign.hasCampaign()) {
  Log.d(TAG, "No campaign is available.");
  stopProcess();
}
```

JoaoRaimundo commented on appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java:
```diff
@@ -72,6 +76,9 @@
   boolean fromBackground = false;
   private Handler handleRetryConnection = new Handler();
   private int connectionRetrys = 0;
```
Retries and not retrys

JoaoRaimundo commented on appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java:
```diff
@@ -72,6 +76,9 @@
   boolean fromBackground = false;
```
Please remove variables that may no longer being used.

JoaoRaimundo commented on appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/WalletUtils.java:
```diff
@@ -7,48 +7,96 @@
+
+  public static Context context;
+
+  public static boolean isDialogVisible() {
```
It should not be the WalletUtils to control if the dialog is already being shown.

JoaoRaimundo commented on appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/WalletUtils.java:
```diff
@@ -7,48 +7,96 @@
+  }
+
+  public static void setContext(Context cont) {
+    context = cont;
```
If this is a utility that provides static method, it should have have fields initialisation. If the idea is to have a class that manages the installation dialog and all the needed components to do so, please consider having a class that is initialises the field without them being static, and the method provided being non static.

On the other hand of the idea is to have a static method and an utility class, please provide the context as parameter to the method that requires it and remove the control the dialog visibility control variable.

JoaoRaimundo commented on appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/WalletUtils.java:
```diff
@@ -7,48 +7,96 @@
     builder.setNegativeButton(R.string.skip, new DialogInterface.OnClickListener() {
       @Override public void onClick(DialogInterface dialogInterface, int i) {
-        dialogVisibleListener.OnDialogVisibleListener(false);
+        dialogInterface.cancel();
```
the method that should be called is dismiss and not cancel. And in this case in both button actions, positive and negative.

JoaoRaimundo commented on appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/WalletUtils.java:
```diff
@@ -7,48 +7,96 @@
+        return activity;
+      }
+    }
+    return null;
```
Since we are already throwing an exception, can we use that to throw an exception instead of null, since the action for both returns (null or exception) is the same.

JoaoRaimundo commented on appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/repository/ResponseCode.java:
```diff
@@ -0,0 +1,17 @@
+package com.asf.appcoins.sdk.ads.repository;
+
+public enum ResponseCode {
+  BILLING_UNAVAILABLE(3), DEVELOPER_ERROR(5), ERROR(6), FEATURE_NOT_SUPPORTED(
```
WE have many response codes that don't make sense in this module, please remove the not used

brunovarelaAptoide committed:
Changes Code review

- Changed dialogVisible hadling
- Removed unused variable

brunovarelaAptoide committed:
Code review changes

- Changed url install wallet
- Minor changes

brunovarelaAptoide committed:
Changes Code Review

- New link to install the wallet.
- Code improvements.

brunovarelaAptoide committed:
Code Improvements

- Code Changes improvements

brunovarelaAptoide committed:
Changes Code Review

- Committed changes code review

brunovarelaAptoide committed:
Code review Changes

-the sdk removes the install notification after wallet is installed

brunovarelaAptoide committed:
Merge remote-tracking branch 'remotes/origin/APPC-1146-SDK-Dialog-Wallet-Install-To-Notification' into APPC-1035-Develop-PoA-standalone-SDK

brunovarelaAptoide committed:
Changes

-Merge changes

JoaoRaimundo commented on appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java:
```diff
@@ -71,7 +75,12 @@
+  private boolean isWalletInstalled;
+  private AppcoinsAdvertisementRepository appcoinsAdvertisementRepository;
+  private AppcoinsAdvertisementConnection appcoinsAdvertisementConnection;
+  private static boolean showPopUpNotification;
```
Remove not used and join the new static variables with the already existing static variables

brunovarelaAptoide committed:
Bug fix

- Bug fix of POA starting 2 times

brunovarelaAptoide committed:
PR Changes

- Removed unusued variable
- Remove unusued interface
- Grouped variables

brunovarelaAptoide committed:
Merge remote-tracking branch 'remotes/origin/APPC-1146-SDK-Notification-withouth-Aptoide-redirect' into APPC-1035-Develop-PoA-standalone-SDK

brunovarelaAptoide committed:
Merge remote-tracking branch 'remotes/origin/dev' into APPC-1035-Develop-PoA-standalone-SDK

brunovarelaAptoide committed:
Changed comentary

- Changed comentary from WalletUtils.java to DialogInstall class
